tests/test_install.py

```diff
@@ -32,54 +32,109 @@ def generate_random_path():
     return "/some/path/to/file%s" % random.randint(100, 200)
 
 
-@pytest.mark.xfail(on_win, reason="binary replacement on windows skipped", strict=True)
-def test_simple():
+@pytest.mark.parametrize("subdir", ["linux-64", "win-64", "noarch"])
+def test_simple(request, subdir):
+    if subdir == "win-64" or (subdir == "noarch" and on_win):
+        request.node.add_marker(
+            pytest.mark.xfail(
+                reason="binary replacement on windows skipped", strict=True
+            )
+        )
```

Comment:
This is a little repetitive, let's move this into an own module level fixture that uses [indirect parametrization](https://docs.pytest.org/en/latest/example/parametrize.html#indirect-parametrization), meaning that the fixture is able to inspect the test function's parameters using the `request.param` parameter.